course5/homework9-benchmark/benchmark.cc

```diff
@@ -0,0 +1,48 @@
+#include <string>
+#include <cstring>
+#include <benchmark/benchmark.h>
+#include <iostream>
+
+bool is_greater_string(const std::string& a, const std::string& b) { return a>b; }
+bool is_greater(const char* a, const char* b) { return std::strcmp(a, b) > 0;}
+
+static void is_greater_bench(benchmark::State& state)
+{
+    std::string a = "a";
+    std::string z = "z";
+    std::string input1 = a;
+    std::string input2 = z;
+    for(int i = 0; i < state.range(0); i++)
+    {
+         input1 += a;
+         input2 += z;
+    }
+    const char* inputchar1 = input1.c_str();
+    const char* inputchar2 = input2.c_str();
+
+    while(state.KeepRunning())
+        benchmark::DoNotOptimize( is_greater(inputchar1, inputchar2 ));
+}
+
+static void is_greater_string_bench(benchmark::State& state)
+{
+    std::string a = "a";
+    std::string z = "z";
+    std::string input1 = a;
+    std::string input2 = z;
+    for(int i = 0; i < state.range(0); i++)
+    {
+         input1 += a;
+         input2 += z;
+    }
+    const char* inputchar1 = input1.c_str();
+    const char* inputchar2 = input2.c_str();
+
+    while(state.KeepRunning())
+        benchmark::DoNotOptimize( is_greater_string(inputchar1, inputchar2) );
+}
```

Comment:
Good!
The common part in both cases could be streamlined-out in some way. It's the same code.